src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs

```diff
@@ -91,6 +91,7 @@ public static class ResolverUtility
                                                     Strings.Error_PackageNotFoundWhenExpected,
                                                         match.Provider.Source,
                                                     ex.PackageIdentity.ToString());
+                        context.Logger.LogError(message);
```

Comment:
Thanks to @nkolev92 for providing pointers to the code in https://github.com/NuGet/Home/issues/13460#issuecomment-2130525626. The detailed information really helped to understand the root cause of the bug.

There was one statement in that linked comment that I thought I could explain here:

> Not sure why InvalidCacheProtocolException in particular gets ignored.

From what I understand, looking at the code, InvalidCacheProtocolException is ignored [here](https://github.com/NuGet/NuGet.Client/blob/a7131d98a28b0be1d21a125b5660fc8782c27ace/src/NuGet.Core/NuGet.Commands/RestoreCommand/SourceRepositoryDependencyProvider.cs#L376-L387) because the caller `ResolverUtility.FindLibraryEntryAsync` has logic to try downloading the package twice. The first time when [PackageNotFoundProtocolException](https://github.com/NuGet/NuGet.Client/blob/dc2e3b27170c779e3c01a7306504fb83cb4ef147/src/NuGet.Core/NuGet.Protocol/Utility/FindPackagesByIdNupkgDownloader.cs#L82-L88) is thrown, the code in the first catch block at the caller invalidates the cache and does a retry. That's why the exception is not logged at the callee but instead logged at the caller.

Reply:
This change is missing a log code. 
It should've probably been NU1301 like the other ones, or potentially a new one.

Reply:
Thinking about it more, this scenario is distinct enough, where I think it could use a separate code.